### DISCUSSION TITLE
LGTM - a code analysis platform for identifying vulnerabilities and preventing them from reaching production.

### DISCUSSION THREAD

**Discussion by gastaldi** [quarkiverse/quarkiverse#99] [Team Posts]
[LGTM](https://lgtm.com/)  is a code analysis platform for finding zero-days and preventing critical vulnerabilities. We have [enabled the application](https://github.com/marketplace/lgtm) in the Quarkiverse organization to build PRs but only for a few repositories for the time being.

If you're an extension maintainer and wish to enable LGTM in your Quarkiverse repository, send a PR in the [quarkiverse-devops repository](https://github.com/quarkiverse/quarkiverse-devops#installing-applications) 

NOTE: The branch must be created in the same repository, it won't work in a separate fork (@quarkiverse/quarkiverse-members should be able to create new branches there) 
 
Here is how the PR should look like: https://github.com/quarkiverse/quarkiverse-devops/commit/c9382dd790a3dea36274256251f4c3d9573b28a8

**Comment by gastaldi** [comment]
BTW your repository needs a `.lgtm.yml` file in the root. Here is an [example](https://github.com/quarkiverse/quarkus-prettytime/blob/main/.lgtm.yml)

**Comment by gastaldi** [comment]
I just learned that the environment running the LGTM analysis does not support Docker (when the `docker-maven-plugin` is used, for example), so make sure the build that runs there does not attempt to initialize it

**Comment by jtama** [comment]
Will it just run on MR ?

**Comment by gastaldi** [comment]
Yes, when a new PR is opened against your repo it should be there as an additional check

**Comment by gastaldi** [comment]
LGTM will be shutdown in December 2022: https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/, the recommendation is to use GitHub Code Scanning instead

**Comment by shumonsharif** [comment]
@gastaldi Just wanted to make sure I understood the link correctly - no immediate action is needed on our part, correct?  GitHub will submit PRs in October to each repo for the migration from LGTM to GitHub Code Scanning, and once we merge that PR in, we would just go ahead and disable LGTM?

**Comment by gastaldi** [comment]
@shumonsharif yes, that's correct.
